apps/search/js/search.js

```diff
@@ -138,38 +138,35 @@
         Object.keys(providers).forEach((providerKey) => {
           var provider = providers[providerKey];
 
-          // If suggestions are disabled, only use local providers
-          if (this.suggestionsEnabled || !provider.remote) {
+          var fetchRemote =
+            UrlHelper.isURL(input) ||
+            msg.data.isPrivateBrowser ||
+            !this.suggestionsEnabled;
 
-            if (provider.remote) {
-              // Do not send full URLs to remote providers
-              // or when inside a private browser.
-              if (UrlHelper.isURL(input) || msg.data.isPrivateBrowser) {
-                return;
-              }
-
-              // Show the loading element when searching remote providers.
-              this.loadingElement.classList.add('loading');
-            }
+          if (provider.remote && !fetchRemote) {
+            return;
+          }
 
-            if (provider.name === 'Suggestions') {
-              var toShow = input.length > 2 &&
-                this.toShowNotice &&
-                this.suggestionNotice.hidden &&
-                navigator.onLine;
-              if (toShow) {
-                this.suggestionNotice.hidden = false;
-              }
-            }
+          // Show the loading element when searching remote providers.
+          this.loadingElement.classList.add('loading');
 
-            provider.search(input).then((results) => {
-              this.collect(provider, results);
-            }).catch((err) => {
-              if (provider.remote) {
-                this.loadingElement.classList.remove('loading');
-              }
-            });
+          if (provider.name === 'Suggestions') {
+            var toShow = input.length > 2 &&
+              this.toShowNotice &&
+              this.suggestionNotice.hidden &&
+              navigator.onLine;
+            if (toShow) {
+              this.suggestionNotice.hidden = false;
+            }
           }
+
+          provider.search(input, fetchRemote).then((results) => {
+            this.collect(provider, results);
+          }).catch((err) => {
+            if (provider.remote) {
+              this.loadingElement.classList.remove('loading');
```

Comment:
Because provider.remote is false for the suggestions provider, won't it be the case that this line never gets reached and the loading animation will continue to show?